### DISCUSSION TITLE
[QUESTION] How to add fragment metadata with kvssink Gstreamer plugin

### DISCUSSION THREAD

**Discussion by sinanliuAmazon** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1001] [Q&A] [answered]
A one liner description about the use case and what you are trying to achieve
Is it possible to add fragment metadata to the stream using kvssink element? If so can you provide a example?
** Logging **
Add relevent SDK logging.  IMPORTANT NOTE:  Please make sure to NOT share AWS access credentials under any circumstance!  Please make sure they are not in the logs.

** Any design considerations/constraints **
Explain in detail how you would like to integrate our SDK into your solution 

** If you would not like to open an issue to discuss your solution in open-platform, please email your question to kinesis-video-support@amazon.com **

**Comment by disa6302** [comment, marked as answer]
@sinanliuAmazon ,

There isnt a way to set metadata with command line, but you can set it up in an application that uses kvssink. You would have to use `gst_element_send_event`. The event name would be `kvs-add-metadata`. 

This portion of code should give you details on what kvssink expects: https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/0888c666efbe7f20891a86c9e64a677f73e33b17/kinesis-video-gstreamer-plugin/src/gstkvssink.cpp#L947-L976

We do not have a sample for this at the moment and we can look into, but hope this helps in the meantime.

**Reply by sinanliuAmazon** [reply]
Great, thank you very much for the prompt response, will try it out.

**Reply by disa6302** [reply]
Oh, and before that, you can create your event using `gst_event_new_custom` (https://gstreamer.freedesktop.org/documentation/gstreamer/gstevent.html?gi-language=c#gst_event_new_custom). This way you can populate your GstStructure which kvssink expects.

**Comment by sinanliuAmazon** [comment]
Tested, sending `kvs-add-metadata` event to the kvssink element works, thanks!

**Comment by richardgavel-ordinaryexperts** [comment]
We are using gst-launch-1.0 directly, vs embedding the gstreamer API into our application. Would it be possible to update the sink to have another sink parmeter similar to the stream_tags that injects fragment metadata values right after initiating the stream? Or is there another plugin which can inject events

**Reply by narolski** [reply]
Would appreciate this as well.
